src/lib.rs

```diff
@@ -1022,14 +1021,78 @@ trait OperationTrait {
 
 ////////////////////////
 
+/// Returns a string describing version information of the
+/// TensorFlow library. TensorFlow using semantic versioning.
+pub fn version() -> std::result::Result<String, Utf8Error> {
+  unsafe {
+    CStr::from_ptr(tf::TF_Version()).to_str().map(|s| s.to_string())
+  }
+}
+
+////////////////////////
+
+/// A TensorShape is the shape of a tensor.  A TensorShape may be an unknown
+/// rank, or it may have a known rank with each dimension being known or
+/// unknown.
+#[derive(Debug,Eq,Ord,PartialEq,PartialOrd,Hash,Clone)]
+pub struct TensorShape(Option<Vec<Option<i64>>>);
+
+impl TensorShape {
+  /// Returns the number of dimensions if known, or None if unknown.
+  pub fn rank(&self) -> Option<usize> {
```

Comment:
Renamed to dims().  Out of curiosity, do you know why "rank" is discouraged?  Is it just to keep the jargon to a minimum?